src/lib/components/Instances/InstanceTable.svelte

```diff
@@ -24,20 +24,16 @@
 
 	let selectedInstances = $state<boolean[]>([]);
 	let generatingInstances = $state(false);
-	let metricValues = $state<Record<string, Promise<Tables<'metrics'> | undefined>>>({});
+	let metrics = $state<Record<string, number | undefined>>({});
 	let showPaywall = $state(false);
 	let showDelete = $state(false);
 
-	async function averageMetric(
-		values: Record<string, Promise<Tables<'metrics'> | undefined>>
-	): Promise<number | undefined> {
-		const result = Promise.all(Object.values(values)).then((d) => {
-			const metrics = d.filter((d) => d !== undefined) as Tables<'metrics'>[];
-			if (metrics.length === 0) return undefined;
-			return metrics.reduce((acc, m) => acc + m.metric, 0) / metrics.length;
-		});
-		return result;
-	}
+	let metricValues = $derived(Object.values(metrics).filter((metric) => metric !== undefined));
+	let avgMetric = $derived(
+		metricValues.length === 0
+			? undefined
+			: (metricValues.reduce((a, b) => (a ?? 0) + (b ?? 0), 0) ?? 0) / metricValues.length
```

Comment:
given metricValues only has numbers now, can we remove all these ternary checks?

Reply:
also what should the behavior actually be here. Should the average be of anything with a label or of all instances? idk

Reply:
I think we still need them compiler does not understand that sadly....

I think the behavior like this makes sense because I feel like having an instance without label won't be uncommon and counting that for the metric seems unfair (i.e., this is what I'd expect using it) but open to discuss.